### PR TITLE
Add GroupMembership struct

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Adds struct for storing and modifying GroupMembership
- Adds ability to diff group membership structs and get list of added/removed/updated inboxes
- Serialization to/from protos

### CODE REVIEW

**Comment by neekolas** [comment]
* **#662** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/662?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xmtp_id/src/associations/serialization.rs [hunk 1]
```diff
@@ -200,7 +200,7 @@ fn from_signature_kind_proto(
             signature_text,
             erc1271_signature.signature,
             erc1271_signature.contract_address,
-            erc1271_signature.block_height as u64,
+            erc1271_signature.block_number,
```
Comment (neekolas) [inline xmtp_id/src/associations/serialization.rs:203]: This proto change got picked up along the way

File: xmtp_mls/src/groups/message_history.rs [hunk 2]
```diff
@@ -1,11 +1,13 @@
 use xmtp_proto::{
     api_client::XmtpMlsClient,
     xmtp::mls::message_contents::plaintext_envelope::v2::MessageType::{
-        MessageHistoryRequest as HistoryRequest, MessageHistoryResponse as HistoryResponse,
+        Reply as HistoryResponse, Request as HistoryRequest,
```
Comment (neekolas) [inline xmtp_mls/src/groups/message_history.rs:4]: @tuddman I picked up this proto change of yours. I decided to take the path of least resistance and just alias everything back to the old names. Can I leave it to you to handle the final naming however you want in a subsequent PR.

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs:4]: Sure.  I made the change in part so we'd not need to alias anything.

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs:4]: They'll come in https://github.com/xmtp/libxmtp/pull/660

**Review by tuddman — APPROVED** [review]

**Pull request merged**